src/lib/holocene/button.svelte

```diff
@@ -19,22 +36,20 @@
   export let large: boolean = false;
   export let as: 'button' | 'anchor' = href ? 'anchor' : 'button';
   export let icon: IconName = null;
-  export let classes: string = $$props.class;
-  export let dataCy: string = $$props.dataCy;
   export let count: number = 0;
   export let type: string = 'button';
 </script>
 
 {#if as === 'button'}
   <button
     on:click
-    class="button {variant} {classes}"
+    class="button {variant} {$$props.class}"
     class:selected={active}
     class:large
     class:thin
-    data-cy={dataCy}
     {type}
     {disabled}
+    {...omit($$restProps, 'class')}
```

Comment:
Is there currently a use case where `$$restProps` is needed? 

Just thinking from a design systems standpoint I'm not sure if we want to be passing down unknown attributes. If we need something more extensible it seems like that would indicate we haven't thought through a potential variant or we should be building a custom component 🤔 

> It shares the same optimisation problems as $$props, and is likewise not recommended.

On the other hand... We're already using `$$props` and I know this would likely be useful 😅 Just don't know if it's the _right_ thing to do.

One alternative we could explore would be using objects as props, especially if we want more flexibility for customization of child elements.
```suggestion
export let buttonProps = {}
<div>
  <button
    on:click
    ...
    {...buttonProps}
    ...
```

Reply:
> Is there currently a use case where $$restProps is needed?

This came about because I wanted to write a playwright test in cloud-ui, but couldn't add a `data` attribute or even an `id` to a `<Button />` without first going to add that new prop to the component in Holocene, publishing a new package version, updating the package in cloud-ui, etc. That led to this idea of enabling all Holocene components to accept any valid HTML attribute as a prop. I've seen design systems work similarly to this, because you wouldn't want to maintain a list of valid HTML Attributes for every component in your component library. For example, if we waned Holocene to support `aria` attributes on every component, specifying that for each component would be cumbersome. 

Another pattern I've seen is to create a `<Box />` component, or "primitive", from which all other components are made. The `<Box />` component would take an `as` prop representing the HTML Element it actually renders, so `<Box as="a" />` would render an anchor tag, <Box as="div" />` would render a `<div />` etc. Then this `<Box />` component would maintain this list of valid HTML Attributes and do the prop forwarding itself.

Reply:
> couldn't add a data attribute or even an id to a <Button /> without first going to add that new prop to the component in Holocene

Ah I see. And yeah that isn't ideal 😞 The `<Box />` is an interesting pattern 🤔 
I'll all for giving `$$restProps` a try! Although, I wonder if there's a better approach to this pattern at least...
```js
omit($$restProps, 'class')
```
I'll noodle on it a bit 🍜 Thanks for putting this POC together!

Reply:
@laurakwhit yeah, that's the part of this implementation I'm least OK with, but without that, if a class is passed to a component, it will overwrite our inline component classes entirely :/